test/LogTest.php

```diff
@@ -0,0 +1,231 @@
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Test
+ * @package     OpusTest
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace OpusTest;
+
+use Opus\Log;
+
+class LogTest extends \PHPUnit_Framework_TestCase
+{
+
+    private $logFile;
+
+    public function tearDown()
+    {
+        if (is_resource($this->logFile)) {
+            fclose($this->logFile);
+        }
+
+        parent::tearDown();
+    }
+
+    public function testSetLevel()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(\Zend_Log::DEBUG);
+        $debugMessage = 'Debug level message from testSetLevel';
+        $opusLog->debug($debugMessage);
+
+        $this->assertContains($debugMessage, $this->readLog());
+        $this->assertEquals(\Zend_Log::DEBUG, $opusLog->getLevel());
+    }
+
+    public function testSetLevelForFilterDisabling()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $debugMessage = 'Debug level message from testSetLevelWithNullLevel';
+        $infoMessage = 'Info level message';
+
+        $opusLog->info($infoMessage);
+        $opusLog->debug($debugMessage);
+        $priorContent = $this->readLog();
+
+        $opusLog->setLevel(null);
+
+        $opusLog->debug($debugMessage);
+        $content = $this->readLog();
+
+        $this->assertContains($infoMessage, $priorContent);
+        $this->assertNotContains($debugMessage, $priorContent);
+        $this->assertContains($debugMessage, $content);
+    }
+
+    // To test custom level messages are also working properly when filter is disabled.
+    public function testSetLevelToNullHavingCustomLevels()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->addPriority('TEST', 8);
+        $opusLog->setLevel(null);
+        $opusLog->addPriority('TLEVEL', 9);
+
+        $testMessage = 'Test level created before filter disabled';
+        $opusLog->test($testMessage);
+
+        $tlevelMessage = 'Test level created after filter disabled';
+        $opusLog->tlevel($tlevelMessage);
+
+        $content = $this->readLog();
+
+        $this->assertContains($testMessage, $content);
+        $this->assertContains($tlevelMessage, $content);
+    }
+
+    public function testSetLevelNotModifyingOtherFilters()
+    {
+        $opusLog = $this->getOpusLog();
+        $initialLevel = $opusLog->getLevel(); //Should return INFO
```

Comment:
Okay, and does it return `INFO`? I see the assertion at the end. There is not rule that assertions have to be at the end of test functions. This makes it less readable and wastes time running the test when this basic assumption isn't met.